src/runtime/server/services/utils.ts

```diff
@@ -9,14 +10,16 @@ import { useRuntimeConfig } from '#imports'
  * Get `origin` and fallback to `x-forwarded-host` or `host` headers if not in production.
  */
 export const getServerOrigin = (event?: H3Event): string => {
+  const authRuntimeConfig = useRuntimeConfig().public.auth
+
   // Prio 1: Environment variable
-  const envOrigin = process.env.AUTH_ORIGIN
+  const envOrigin = camelCase(authRuntimeConfig.originEnvKey!, { normalize: true })
```

Comment:
I don't really get the point of this - you are camelCasing the env variable name? In the default case, it becomes
```ts
camelCase('AUTH_ORIGIN', { normalize: true }) // `authOrigin`
```

But this is not useful at all

Reply:
Nuxt waits for env variable with prefix `NUXT_` (by default, this prefix can be changed to a custom prefix). 

When Nuxt receives the variables, it converts them to camlCase by removing the `NUXT_` prefix. 

Here we get the ENV from the `authRuntimeConfig.originEnvKey` variable (whose name we can change to be package-independent)

`NUXT_AUTH_ORIGIN` -> `useRuntimeConfig().authOrigin`.

PS: i noticed, i'm not extracting the variable from ENV, i fixed that

Translated with DeepL.com (free version)

Reply:
Thanks for the comment, that was indeed a mistake